Sources/MUXSDKStatsObjc/MUXSDKMonitor.m

```diff
@@ -189,67 +364,235 @@ - (nullable MUXSDKPlayerBinding *)startMonitoringPlayer:(nonnull AVPlayer *)play
+
 }
 
 - (void)stopMonitoringWithPlayerBinding:(nonnull MUXSDKPlayerBinding *)playerBinding {
-    [self doesNotRecognizeSelector:_cmd];
+    NSString *playerName = playerBinding.playerName;
+    NSString *playerSoftwareName = [_playerSoftwareNamesByPlayerName valueForKey:playerName];
+    if (playerSoftwareName == MUXSDKPlayerSoftwareAVPlayerViewController) {
+        MUXSDKAVPlayerViewControllerBinding *player = [_playerBindingManager.playerBindings valueForKey:playerName];
+        [player dispatchViewEnd];
+        [player detachAVPlayer];
+        [_playerBindingManager.playerBindings removeObjectForKey:playerName];
+    } else if (playerSoftwareName == MUXSDKPlayerSoftwareAVPlayerLayer) {
+        MUXSDKAVPlayerLayerBinding *player = [_playerBindingManager.playerBindings valueForKey:playerName];
+        [player dispatchViewEnd];
+        [player detachAVPlayer];
+        [_playerBindingManager.playerBindings removeObjectForKey:playerName];
+    } else if (playerSoftwareName == MUXSDKPlayerSoftwareAVPlayer) {
+        MUXSDKAVPlayerBinding *player = [_playerBindingManager.playerBindings valueForKey:playerName];
+        [player dispatchViewEnd];
+        [player detachAVPlayer];
+        [_playerBindingManager.playerBindings removeObjectForKey:playerName];
+    }
+    [_playerSoftwareNamesByPlayerName removeObjectForKey:playerName];
+    [_playerBindingManager removeBindingsForPlayerName:playerName];
 }
 
 #pragma mark - Automatic Video Change
 
 - (void)updateAutomaticVideoChangeForPlayerName:(nonnull NSString *)playerName
                                         enabled:(BOOL)enabled {
-    [self doesNotRecognizeSelector:_cmd];
+    MUXSDKPlayerBinding *player = [_playerBindingManager.playerBindings valueForKey:playerName];
+    if (player) {
+        [player setAutomaticVideoChange:enabled];
+    }
 }
 
 #pragma mark - Manual Video Change
 
 - (void)signalVideoChangeForPlayerName:(nonnull NSString *)playerName
                withUpdatedCustomerData:(nullable MUXSDKCustomerData *)customerData {
-    [self doesNotRecognizeSelector:_cmd];
+    MUXSDKCustomerPlayerData *playerData = customerData.customerPlayerData ?: [_playerBindingManager.customerPlayerDataStore playerDataForPlayerName:playerName];
+    MUXSDKCustomerViewData *viewData = [customerData customerViewData];
+    MUXSDKCustomerVideoData *videoData = [customerData customerVideoData];
+    MUXSDKCustomData *customData = [customerData customData];
+
+    if (!(videoData || viewData || customData)) {
+        NSLog(@"MUXSDK-WARNING - Mux failed to signal a video change because no video, view, and custom data was supplied for new video playing on player name: %@", playerName);
+        return;
+    }
+    MUXSDKPlayerBinding *player = [_playerBindingManager.playerBindings valueForKey:playerName];
+    if (player) {
+        [player didTriggerManualVideoChange];
+        [player dispatchViewEnd];
+
+        // FIXME: Previous implementation did a nil check
```

Comment:
I think this change is safe, `MUXSDKDataEvent` properties default to `nil` so skipping a nil check gets the same result.